MontyHallv2/GameModes/ConsoleGame.cs

```diff
@@ -0,0 +1,60 @@
+
+using MontyHallv2.DoorCreation;
+using MontyHallv2.GameShowStaff;
+using MontyHallv2.Interfaces;
+using MontyHallv2.Messages;
+using MontyHallv2.Strategies;
+using MontyHallv2.Validation;
+
+namespace MontyHallv2.GameModes;
+
+public class ConsoleGame : IGameMode
+{
+    private int _playerChoice;
+    private string? _playerSelection;
+    
+    public void PlayerChooseDoor(List<Door> doors)
+    {
+        _playerSelection = GetDoorSelectionFromUser();
+        while (!PlayerChoiceValidator.ValidateUserDoorSelection(_playerSelection))
+        {
+            Dialog.InvalidDoorSelectionInputMessage();
+            _playerSelection = GetDoorSelectionFromUser();
+        }
+
+        if (_playerSelection != null) _playerChoice = int.Parse(_playerSelection) - 1;
+        doors[_playerChoice].PlayerPickedDoor();
+    }
+
+    public void PlayerSwitchOrStayDoor(List<Door> doors)
+    {
+        
+        var strategy = new ToSwitch();
+        _playerSelection = GetSwitchOrStayChoiceFromUser();
+
+        while (!PlayerChoiceValidator.ValidateUserSwitchOrStayChoice(_playerSelection))
+        {
+            Dialog.InvalidSwitchOrStayChoiceInputMessage();
+            _playerSelection = GetSwitchOrStayChoiceFromUser();
+        }
+        
+        if (_playerSelection == "y")
+        {
+            strategy.ToSwitchOrStay(doors);
+        }
```

Comment:
this conditional seems to be on if the player has elected to switch?

but then the `strategy.ToSwitchOrStay(doors);` makes it sound like there's still a choice to be made. would be easier to read if the function call was something closer to `Switch(doors)`